app/repositories/channels.py

```diff
@@ -0,0 +1,205 @@
+            AND write_priv = COALESCE(:write_priv, write_priv)
+            AND auto_join = COALESCE(:auto_join, auto_join)
+    """
+    params = {
+        "read_priv": read_priv,
+        "write_priv": write_priv,
+        "auto_join": auto_join,
+    }
+
+    rec = await app.state.services.database.fetch_one(query, params)
+    assert rec is not None
+    return rec["count"]
+
+
+async def fetch_many(
+    read_priv: Optional[int] = None,
+    write_priv: Optional[int] = None,
+    auto_join: Optional[bool] = None,
+    page: Optional[int] = None,
+    page_size: Optional[int] = None,
+) -> list[dict[str, Any]]:
+    """Fetch multiple channels from the database."""
+    query = f"""\
+        SELECT {READ_PARAMS}
+          FROM channels
+         WHERE read_priv = COALESCE(:read_priv, read_priv)
+            AND write_priv = COALESCE(:write_priv, write_priv)
+            AND auto_join = COALESCE(:auto_join, auto_join)
+    """
+    params = {
+        "read_priv": read_priv,
+        "write_priv": write_priv,
+        "auto_join": auto_join,
+    }
+
+    if page is not None and page_size is not None:
+        query += """\
+            LIMIT :limit
+           OFFSET :offset
+        """
+        params["limit"] = page_size
+        params["offset"] = (page - 1) * page_size
+
+    recs = await app.state.services.database.fetch_all(query, params)
+    return [dict(rec) for rec in recs]
+
+
+async def update(
+    name: str,
+    topic: Optional[str] = None,
+    read_priv: Optional[int] = None,
+    write_priv: Optional[int] = None,
+    auto_join: Optional[bool] = None,
+) -> Optional[dict[str, Any]]:
+    """Update a channel in the database."""
+    query = """\
+        UPDATE channels
+           SET topic = COALESCE(:topic, topic),
+               read_priv = COALESCE(:read_priv, read_priv),
+               write_priv = COALESCE(:write_priv, write_priv),
+               auto_join = COALESCE(:auto_join, auto_join)
+         WHERE name = :name
+    """
+    params = {
+        "name": name,
+        "topic": topic,
+        "read_priv": read_priv,
+        "write_priv": write_priv,
+        "auto_join": auto_join,
+    }
+    await app.state.services.database.execute(query, params)
+
+    query = f"""\
+        SELECT {READ_PARAMS}
+          FROM channels
+         WHERE name = :name
+    """
+    params = {
+        "name": name,
+    }
+    rec = await app.state.services.database.fetch_one(query, params)
+    return dict(rec) if rec is not None else None
+
+
+async def delete(
+    name: str,
+) -> Optional[dict[str, Any]]:
+    """Delete a channel from the database."""
+    query = f"""\
+        SELECT {READ_PARAMS}
+            FROM channels
+              WHERE name = :name
+    """
+    params = {
+        "name": name,
+    }
+    rec = await app.state.services.database.fetch_one(query, params)
+    if rec is None:
+        return None
+
+    query = """\
+        DELETE FROM channels
+         WHERE name = :name
```

Comment:
this looks super bad, but it's what i'm doing in other places
```suggestion
        DELETE FROM channels
              WHERE name = :name
```

Reply:
i might consider
```py
        DELETE
          FROM channels
         WHERE name = :name
```